Electronic Colloquium on Computational Complexity.js

```diff
@@ -1,94 +1,122 @@
 {
 	"translatorID": "09a9599e-c20e-a405-d10d-35ad4130a426",
 	"label": "Electronic Colloquium on Computational Complexity",
-	"creator": "Jonas Schrieb",
-	"target": "^https?://eccc\\.weizmann\\.ac\\.il/",
+	"creator": "Jonas Schrieb and Morgan Shirley",
+	"target": "^https?://eccc\\.weizmann\\.ac\\.il/(title|year|keyword|report)",
 	"minVersion": "1.0.0b3.r1",
 	"maxVersion": "",
 	"priority": 100,
 	"inRepository": true,
 	"translatorType": 4,
-	"browserSupport": "gcsib",
-	"lastUpdated": "2017-01-05 17:11:41"
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2023-05-31 20:54:48"
 }
 
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2023 Jonas Schrieb and Morgan Shirley
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
 function detectWeb(doc, url) {
-	var singleRe   = /^https?:\/\/eccc\.weizmann\.ac\.il\/report\/\d{4}\/\d{3}/;
 	var multipleRe = /^https?:\/\/eccc\.weizmann\.ac\.il\/(title|year|keyword)\//;
-	if (singleRe.test(url)) {
-		return "report";
-	} else if (multipleRe.test(url)) {
+	if (multipleRe.test(url)) {
 		return "multiple";
 	}
+	else if (ZU.fieldIsValidForType('title', 'preprint')) {
```

Comment:
Could you use the `const` approach as e.g. here:
https://github.com/zotero/translators/blob/master/OSF%20Preprints.js#L39

a) it doesn't require you to duplicate code quite as much and b) it's what we've done pretty consistently in other translators, so we can remove the report code more quickly once it's no longer needed

Reply:
Done. The original code was based on the arXiv.org translator; that one should probably be changed as well.